R/plotPercentiles.R

```diff
@@ -0,0 +1,107 @@
+#' Comparison plots show user selected variables obtained from different scenario
+#' runs
+
+#' @author Tonn Rueter
+#' @param df `quitte` style data frame containing all variables for each scenario
+#' @param scenarios Character vector contains names of the desired scenarios. If none, all scenarios will be displayed
+#' @param variables Character vector contains names of the desired variables. If none, all variables will be displayed
+#' @importFrom dplyr filter mutate vars
+#' @importFrom reshape2 melt
+#' @importFrom stringr str_extract
+#' @importFrom tidyr pivot_wider
+#' @importFrom ggplot2 ggplot geom_line geom_ribbon facet_wrap facet_grid theme ylab
+#' @export
+plotPercentiles <- function(df, scenarios = NULL, variables = NULL) {
+
+  # In the quitte data frame all perenctiles are given as individual variables
+  # Manipulate input data frame such that all percentiles of a given quantity
+  # are transformed to individual columns. Variable names in the quitte data
+  # frame follow the format "Any|Variable|5.0th Percentile". The regular
+  # expressions below divide the variable name into the prefix and the
+  # percentile specifier
+  data <- df %>%
+    mutate(
+      "percentile" = stringr::str_extract(.data$variable, "[^\\|]+?$"),
+      "variable"   = gsub("\\|[^\\|]+$", "", .data$variable)
+    ) %>%
+    pivot_wider(
+      names_from = "percentile",
+      values_from = "value"
+    )
+
+  # Check which scenarios/variabes are available
+  uniqueScenarios <- unique(data$scenario)
+  uniqueVariables <- unique(data$variable)
+
+  # Check which function parameters have been provided and default to unique
+  # values from the data frame in case none have
+  theseScenarios <- if (is.null(scenarios)) {
+    uniqueScenarios
+  } else if (allItemsAvailable(scenarios, uniqueScenarios, warn = TRUE)) {
+    scenarios
+  } else {
+    stop("Provided scenario is missing in data")
+  }
+  theseVariables <- if (is.null(variables)) {
+    uniqueVariables
+  } else if (allItemsAvailable(variables, uniqueVariables, warn = TRUE)) {
+    variables
+  } else {
+    stop("Provided variable is missing in data")
+  }
+
+  # Set up the plot
+  p <- ggplot()
+
+  # Fill plot by filtering for the requested variables and scenarios
+  for (thisVariable in theseVariables) {
+    for (thisScenario in theseScenarios) {
+      plotData <- filter(data, .data$variable == thisVariable & .data$scenario == thisScenario)
+      p <- p +
+        geom_line(
+          data = plotData, aes(x = .data$period, y = get("50.0th Percentile"))
```

Comment:
This is another assumption about the data you must state explicitly in the documentation. As of now, you expect variables for 5,33,50,67,95 the percentile with very specific names in the input data, so add it to the documentation.

(In the future, we might want to make this more flexible. But this will be for the stakeholders - Oliver and Gabriel - to decide)